### PR TITLE
Feature/lp 16 listing page development

### PR DESCRIPTION
**Pull request by allisono-hs**
- table functionality
- table UI

need to fix 
- warnings
- icons on table
- spacing of right side of nav bar

### CODE REVIEW

**Comment by YogirajA** [comment]
1. Please fix some of the unverified commits. Let me know if you want me to flag them
2. The front end should handle errors coming from the API and display them at the message on top. The error bod looks like this. 
ApiError
3. The names are looking incorrect. It can be handled in the backend as well if that makes life easier. 
FirstName

**Comment by allisono-hs** [comment]
Took care of issues 2-5

**Review by YogirajA — CHANGES_REQUESTED** [review]

File: src/API/LeadershiProfileAPI.Tests/ProfileControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        [Fact]
+        public async Task ShouldReturnListOfTeacherProfiles()
+        {
+            var mockHttpClientFactory = new MockHttpClientFactory();
+            var mockLogger = new MockLogger();
+
+            var controller = new ProfileController(mockLogger, mockHttpClientFactory);
+
+            var result = (await controller.GetDirectory("1",null,null,null)).Profiles.ToList();
```
[the remark this answers is not in the document]
Comment (allisono-hs) [reply src/API/LeadershiProfileAPI.Tests/ProfileControllerTests.cs:22]: Should we build more tests or just fix this test?

File: src/API/LeadershipProfileAPI/Controllers/ProfileController.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            var staffsRequest = new HttpRequestMessage(HttpMethod.Get,
+                QueryHelpers.AddQueryString($"{Constants.VersionUriFragment}/ed-fi/staffs", query));
+            
+            var response = await GetApiResponse<IList<Models.TeacherProfile>>(client, staffsRequest).ConfigureAwait(false);
+
+            var staffCount = Convert.ToInt32(response.headers.GetHeader("Total-Count"));
+
+            await GetAssociations(client, response.response).ConfigureAwait(false);
+
+            foreach (var teacherProfile in response.response)
```
[the remark this answers is not in the document]
Comment (allisono-hs) [reply src/API/LeadershipProfileAPI/Controllers/ProfileController.cs:59]: seperate ticket?

File: src/API/LeadershipProfileAPI/Controllers/Extensions.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                return 0;
+
+            return (DateTime.Today.Year - Convert.ToDateTime(org.beginDate).Year);
+        }
+
+        public static string GetLocation(this IList<Models.Address> addresses)
+        {   
+            var address = addresses.FirstOrDefault();
+
+            return address == null ? "Default Location" : $"{address.city}, {address.stateAbbreviationDescriptor.Split('#')[1]}";
```
[the remark this answers is not in the document]
Comment (allisono-hs) [reply src/API/LeadershipProfileAPI/Controllers/Extensions.cs:28]: with the https://api.ed-fi.org/ there's no addresses to see and test this

File: src/Web/.gitignore [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
[the remark this answers is not in the document]
Comment (allisono-hs) [reply src/Web/.gitignore:1]: done

File: src/Web/public/robots.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
```
[the remark this answers is not in the document]
Comment (allisono-hs) [reply src/Web/public/robots.txt:1]: deleted

File: src/Web/src/setupTests.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
+// expect(element).toHaveTextContent(/react/i)
+// learn more: https://github.com/testing-library/jest-dom
+import '@testing-library/jest-dom';
```
[the remark this answers is not in the document]
Comment (allisono-hs) [reply src/Web/src/setupTests.js:5]: wasnt planning on it so deleting it

File: src/API/LeadershipProfileAPI/Controllers/ProfileController.cs [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+            {
+                teacherProfile.Location = teacherProfile.Addresses.GetLocation();
+            }
+
+            var currentPageProfiles = response.response.AsQueryable();
+
+            if (search != null)
+                currentPageProfiles = currentPageProfiles
+                    .Where(x => x.FirstName.ToLowerInvariant().Contains(search) || x.LastName.ToLowerInvariant().Contains(search))
```
[the remark this answers is not in the document]
Comment (allisono-hs) [reply src/API/LeadershipProfileAPI/Controllers/ProfileController.cs:68]: commented out

File: src/API/LeadershipProfileAPI/Controllers/Models.cs [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+{
+    public class Models
+    {
+
+        public class TeacherProfile
+        {
+            [JsonPropertyName("id")] public string Id { get; set; }
+            [JsonPropertyName("firstName")] public string FirstName { get; set; }
+            [JsonPropertyName("middleName")] public string MiddleName { get; set; }
+            [JsonPropertyName("lastSurname")] public string LastName { get; set; }
```
[the remark this answers is not in the document]
Comment (allisono-hs) [reply src/API/LeadershipProfileAPI/Controllers/Models.cs:17]: done!

File: src/Web/README.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
```
[the remark this answers is not in the document]
Comment (allisono-hs) [reply src/Web/README.md:1]: done!

File: src/Web/src/App.test.js [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
```
[the remark this answers is not in the document]
Comment (allisono-hs) [reply src/Web/src/App.test.js:1]: no, deleted

File: src/API/LeadershiProfileAPI.Tests/ProfileControllerTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (YogirajA) [reply src/API/LeadershiProfileAPI.Tests/ProfileControllerTests.cs:23]: added more tests and this test works now

File: src/API/LeadershipProfileAPI/Controllers/ProfileController.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (YogirajA) [reply src/API/LeadershipProfileAPI/Controllers/ProfileController.cs:59]: We will address this in the future.

File: src/API/LeadershipProfileAPI/appsettings.Development.json [hunk 11]
```diff
@@ -0,0 +1,9 @@
+{
+  "Logging": {
```
[the remark this answers is not in the document]
Comment (YogirajA) [reply src/API/LeadershipProfileAPI/appsettings.Development.json]: This is good to keep.

File: src/API/LeadershipProfileAPI/Controllers/Extensions.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (YogirajA) [reply src/API/LeadershipProfileAPI/Controllers/Extensions.cs]: fixed

**Review by YogirajA — APPROVED** [review]

**Pull request merged**